### ISSUE TITLE
Namespace name?

### ISSUE THREAD

**Issue by eliasdaler** [eliasdaler/MetaStuff#11]
Namespace 'meta' is way too common, so it can lead to some name conflicts. 'metastuff', maybe? I won't change anything for now, so I'll leave it open for discussion.

**Comment by Bl4ckb0ne** [comment]
I propose the following : 
``` c++
namespace ms
```
for MetaStuff, `metastuff` is way too long for a namespace.

**Comment by eliasdaler** [comment]
It looks like "ms" namespace is not used by any major library, but still, it seems kinda generic. And you can do this:

```c++
namespace ms = metastuff```

**Comment by Bl4ckb0ne** [comment]
Then I don't see any objection about using `metastuff` as namespace

**Comment by eliasdaler** [comment]
Yeah, now it looks like a good name. I'm still not 100% sure about MetaStuff name, as it's a bit too generic as well (it would better suit a general purpose meta-programming library), but I think I'll go with `metastuff` if I don't find something better.

**Comment by Bl4ckb0ne** [comment]
MetaStuff as the name of the library?

**Comment by eliasdaler** [comment]
Yes, maybe library needs a better name.

**Comment by aggsol** [comment]
How about "SelfReflect" or the like? More than a better name it needs documentation, I have troubles figuring out all the tricks.

**Comment by eliasdaler** [comment]
Yes, SelfReflect sounds pretty good! I like that people can also do something like:

```namespace sr = SelfReflect```

Docs - yeah... I just put it as a proof of concept at first. Didn't really think people are going to use it in real projects.

If you have some questions - feel free to make issues or mail me. For now, example and [this article](https://eliasdaler.github.io/meta-stuff/) are the main sources. In the future I'll be able to write docs, yes.

**Comment by ninkibah** [comment]
Last Christmas, I was playing around with reflection ideas, and created a little library (not on Github, and probably won't be) called Spiegel, which is German for mirror. Maybe Zerkalo, which Google translate tells me is Russian for mirror?

It certainly won't clash with much C++ code out there.
